File: frontend/types/paper.ts
/**
 * # Paper Types
 * 
 * TypeScript types for papers, chunks, embeddings, and paper-related data structures.
 */
export interface Paper {
  id: string;
  title: string;
  authors: string[];
  abstract?: string;
  pdf_path?: string;
  pdf_url?: string;
  doi?: string;
  publication_date?: string;
  upload_date: string;
  workspace_id: string;
  user_id: string;
  status: PaperStatus;
  metadata?: Record<string, any>;
}

export enum PaperStatus {
  PROCESSING = "processing",
  READY = "ready",
  ERROR = "error",
}

export interface Chunk {
  id: string;
  paper_id: string;
  chunk_index: number;
  page_number: number;
  text: string;
  start_char: number;
  end_char: number;
  vector_id: string;
  metadata?: Record<string, any>;
}

export interface PaperListResponse {
  papers: Paper[];
  total: number;
}

File: backend/app/models/database/paper.py
"""
# Paper Database Model

## What it does:
SQLAlchemy model for Paper table. Defines database schema for papers.

## How it works:
- Defines Paper table with columns
- Relationships to other tables (workspace, chunks, user)
- Provides ORM interface for paper operations

## What to include:
- Paper model with columns:
  - id: UUID (primary key)
  - title: String
  - authors: JSON (list of authors)
  - abstract: Text (optional)
  - pdf_path: String (file path)
  - pdf_url: String (URL to PDF, optional)
  - doi: String (optional)
  - publication_date: Date (optional)
  - upload_date: DateTime
  - workspace_id: UUID (foreign key)
  - user_id: UUID (foreign key, uploader)
  - status: Enum (processing, ready, error)
  - metadata: JSON (additional metadata)
- Relationships:
  - workspace: Many-to-one with Workspace
  - user: Many-to-one with User
- Indexes on: workspace_id, user_id, upload_date
"""

